src/Linker.DefineFunction.tt

```diff
@@ -0,0 +1,310 @@
+﻿<#@ template debug="false" hostspecific="false" language="C#" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ import namespace="System.Globalization" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Linq" #>
+<#@ output extension=".cs" encoding="us-ascii" #>
+<#
+// Note: We specify ASCII as output encoding above to prevent different UTF-8 BOM behavior with VS and dotnet-t4.
+#>
+// <auto-generated>
+// This file is autogenerated from a T4 text template (Linker.DefineFunction.tt).
```

Comment:
There appears to be some shared code between this and the other template.

Do you think it would be possible to perhaps split the templates up into multiple files which could be included with the [include directive](https://learn.microsoft.com/en-us/visualstudio/modeling/t4-include-directive) so as to minimize the duplication?

I haven't looked too closely on exactly what is duplicated, so perhaps there are enough subtle differences between these two templates as to make that more trouble than it's worth.